### ISSUE TITLE
Recent Searches feature (Combobox, localstorage, and more)

### ISSUE THREAD

**Issue by zackkrida** [WordPress/openverse-frontend#479]
## Description

We plan to add a new feature to show the user their 'recent searches', in a box below the searchbar. I originally thought this would be a combobox feature, but we shouldn't actually need to implement that until we're doing search _suggestions_, where the user input determines the contents of the box. In this use case we're just showing the nth-most recent searches.

> A preview of the recent searches

![image](https://user-images.githubusercontent.com/6351754/144633420-a3aeadac-187a-455d-9b97-1358d22b7046.png)

Figma Link: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A29056

## Dev Tasks

- [ ] Implement the suggestions box
  - [ ] Popover for the desktop (above `lg`)
  - [ ] Full page modal with a search bar for mobile screens
- [x] Create a composable for storing recent searches
  - [x] Store user recent searches in indexedDb (see https://github.com/WordPress/openverse-frontend/issues/479#issuecomment-1252941676 for justification)
  - [x] Enable clearing of the recent searches in indexedDb

## Implementation questions

- Do we need a way to let users opt-out of this? (maybe this can wait)
- How many items do we want to show? (4 in the mockup currently) I imagine the search bar will populate and the search will be instantly submitted.
- What will happen when the user clicks on one of the recent searches?

## Design decisions (cc @panchovm)

- [x] Add this to the Figma component library (it's currently only in the page mockups)
- [x] Implement a 'clear' button (will update the mockups for this)
- [x] What will the hover/focus styles look like for the choices?

**Comment by dhruvkb** [comment]
I think it'll be nice to have a way to let the users know that this info is 100% local and is not shared with us? It might help a lot to assuage fears of privacy invasion and persuade users to let this feature remain enabled.

**Comment by zackkrida** [comment]
I was thinking about that too. Maybe some small disclaimer text at the bottom of the popup? cc @panchovm 

Something like "Openverse does not store your recent searches; this information is kept locally in your browser" or something like that

**Comment by panchovm** [comment]
I really like @dhruvkb's idea. I will come back with a proposal. This also reminds me, do we have a privacy terms section for Openverse in the [privacy page](https://wordpress.org/about/privacy/)?

**Comment by obulat** [comment]
We plan to implement the Recent Searches feature as part of the [Create new header](https://github.com/WordPress/openverse-frontend/milestone/18) milestone. This is the update on the state of the design for it.

I've found the following newer mockups for the Recent searches feature:
**Desktop**: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Audio-Release?node-id=1001%3A16848
**Mobile**: https://www.figma.com/file/KxkZmBNk218V2Fke1GeYCu/New-header-component?node-id=485%3A69636

Here are the screenshots (please note that they can be outdated, and always check in the Figma mockups):

<img width="410" alt="Screen Shot 2022-09-13 at 12 50 12 PM" src="https://user-images.githubusercontent.com/15233243/189870882-9a9c7802-2ff2-47ee-8d90-fe81003422f6.png">

<img width="125" alt="Screen Shot 2022-09-13 at 12 50 31 PM" src="https://user-images.githubusercontent.com/15233243/189870894-f80d4d0a-c0cb-43aa-8209-6831e88fa46a.png">

Are the desktop mockups current, @panchovm?

Some of the design questions from @zackkrida in the first message of this issue seem to be answered in those mockups:
- [x] Implement a 'clear' button: both the desktop and the mobile mockups have a 'clear' button, in active state in the mobile mockup and resting state in the desktop mockup.
- [ ] What will the hover/focus styles look like for the choices? - Are the mobile styles for hover/focus the same as the desktop styles?

**Comment by obulat** [comment]
The summary of the answers for the Implementation questions above:

>  Do we need a way to let users opt out of this?

In the current milestone, we do not let the users opt out of this, but we add a disclaimer to inform the users that the data is stored locally and not sent anywhere: "Openverse does not store your recent searches. This information is kept locally in your browser."

>   How many items do we want to show?

4, as in both mobile and desktop mockups.

>   What will happen when the user clicks on one of the recent searches?

I agree with @zackkrida's suggestion: _I imagine the search bar will populate and the search will be instantly submitted._

As for implementation details, we **can** re-use the existing popover/dialog implementations from the Mobile menu/content switcher. The main difference here is that we might make the mobile dialog smaller than full height to show the search bar at the top of it. Another alternative is duplicating the search bar at the top of the full-height mobile modal.

**Comment by sarayourfriend** [comment]
It'd be nice to use IndexedDB here rather than localStorage. Given that the retrieval of the information would probably happend _during input_ it would be great not to have the process blocked on local storage retrievals, especially on low spec devices where the response may take longer to come back.

**Comment by zackkrida** [comment]
@sarayourfriend I actually think localstorage could be more appropriate:

1. This data is really small, it's only a few strings, and we could just load it when the header loads rather than on-demand at input time
2. The localstroage api is much simpler and therefore it might be easier to write _and_ to migrate away from if we switch to a session cookie-based approach later on.

**Comment by stacimc** [comment]
@zackkrida I see you opened #1865 -- should this issue be assigned to you?
